projects/transformers/experiments/bertitos.py

```diff
@@ -0,0 +1,222 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""
+These configs serve to provide a smaller version of Bert for rapid prototyping.
+"""
+
+from copy import deepcopy
+
+from .base import debug_bert
+from .bert_replication import bert_100k
+
+"""
+Reduced sized versions of Bert based off the repo:
+https://github.com/google-research/bert (See this link for GLUE scores.)
+
+The batch size of 128 was maximized under the constraint that training fit on
+a single p3.2xlarge GPU.
+
+Both the train and validation sets are comprised of 1% Wikipedia and 8% of Book Corpus
+for a total of 1,132,942 text samples once tokenized with a max_seq_length of 128.
+This is roughly 8.7 gigabytes of tokenized data total.
+See ../notebooks/create_little_text_dataset.ipynb
+"""
+
+#
+# Params and Results:
+#
+# |--------------------------------------------------------------------------|
+# | model           | num params | train loss  | eval loss      | perplexity |
+# |-----------------|:----------:|:-----------:|:--------------:|:----------:|
+# | small_bert_100k | 27,523,072 | 2.951       | 3.861          | 14.532     |
+# | small_bert_50k  | 27,523,072 | 3.373       | 4.417          | 21.386     |
+# | mini_bert_100k  | 10,615,808 | 3.669       | 4.773          | 27.342     |
+# | mini_bert_50k   | 10,615,808 | 4.464       | 5.815          | 56.309     |
+# | tiny_bert_100k  |  4,124,928 | 4.566       | 5.802          | 55.79276   |
+# | tiny_bert_50k   |  4,124,928 | 5.990       | 8.367          | 330.234    |
+# |--------------------------------------------------------------------------|
+#
+#
+# Training Times:
+#
+# |---------------------------------|
+# | model           | training time |
+# |-----------------|:-------------:|
+# | small_bert_100k | ~8 hrs        |
+# | mini_bert_100k  | ~6 hrs        |
+# | tiny_bert_100k  | ~2 hrs 20 min |
+# |---------------------------------|
+# (reported using a one p3.2xlarge)
+#
+
```

Comment:
It would be useful for quick reference to add one line per method with a summary of each bertito - num attention heads, hidden layers, hidden size and intermediate size. That could be part of the second table for example.